src/datapilot/core/platforms/dbt/insights/checks/base.py

```diff
@@ -0,0 +1,27 @@
+from abc import abstractmethod
+from typing import Tuple
+
+from datapilot.core.platforms.dbt.insights.base import DBTInsight
+
+
+class ChecksInsight(DBTInsight):
+    TYPE = "Checks"
+
+    @abstractmethod
+    def generate(self, *args, **kwargs) -> dict:
+        pass
+
+    @classmethod
+    def has_all_required_data(cls, has_manifest: bool, has_catalog: bool, **kwargs) -> Tuple[bool, str]:
+        """
+        Check if all required data is available for the insight to run.
+        :param has_manifest: A boolean indicating if manifest is available.
+        :return: A boolean indicating if all required data is available.
+        """
+        if not has_manifest:
+            return False, "Manifest is required for insight to run."
+
+        if not has_catalog:
```

Comment:
Most checks only require manifest. I think by default we should only check for manifest